src/pages/home/report/ReportActionsList.js

```diff
@@ -0,0 +1,189 @@
+        hasOutstandingIOU: PropTypes.bool,
+    }).isRequired,
+
+    /** Sorted actions prepared for display */
+    sortedReportActions: PropTypes.arrayOf(PropTypes.shape({
+        /** Index of the action in the array */
+        index: PropTypes.number,
+
+        /** The action itself */
+        action: PropTypes.shape(reportActionPropTypes),
+    })).isRequired,
+
+    /** The sequence number of the most recent IOU report connected with the shown report */
+    mostRecentIOUReportSequenceNumber: PropTypes.number,
+
+    /** Are we loading more report actions? */
+    isLoadingReportActions: PropTypes.bool.isRequired,
+
+    /** Callback executed on list layout */
+    onLayout: PropTypes.func.isRequired,
+
+    /** Callback executed on scroll */
+    onScroll: PropTypes.func.isRequired,
+
+    ...withDrawerPropTypes,
+    ...windowDimensionsPropTypes,
+};
+
+const defaultProps = {
+    personalDetails: {},
+    mostRecentIOUReportSequenceNumber: undefined,
+};
+
+class ReportActionsList extends React.Component {
+    constructor(props) {
+        super(props);
+        this.renderItem = this.renderItem.bind(this);
+        this.renderCell = this.renderCell.bind(this);
+        this.keyExtractor = this.keyExtractor.bind(this);
+    }
+
+    /**
+     * Calculates the ideal number of report actions to render in the first render, based on the screen height and on
+     * the height of the smallest report action possible.
+     * @return {Number}
+     */
+    calculateInitialNumToRender() {
+        const minimumReportActionHeight = styles.chatItem.paddingTop + styles.chatItem.paddingBottom
+            + variables.fontSizeNormalHeight;
+        const availableHeight = this.props.windowHeight
+            - (styles.chatFooter.minHeight + variables.contentHeaderHeight);
+        return Math.ceil(availableHeight / minimumReportActionHeight);
+    }
+
+    /**
+     * Create a unique key for Each Action in the FlatList.
+     * We use a combination of sequenceNumber and clientID in case the clientID are the same - which
+     * shouldn't happen, but might be possible in some rare cases.
+     * @param {Object} item
+     * @return {String}
+     */
+    keyExtractor(item) {
+        return `${item.action.sequenceNumber}${item.action.clientID}`;
+    }
+
+    /**
+     * Do not move this or make it an anonymous function it is a method
+     * so it will not be recreated each time we render an item
+     *
+     * See: https://reactnative.dev/docs/optimizing-flatlist-configuration#avoid-anonymous-function-on-renderitem
+     *
+     * @param {Object} args
+     * @param {Object} args.item
+     * @param {Number} args.index
+     *
+     * @returns {React.Component}
+     */
+    renderItem({
+        item,
+        index,
+    }) {
+        const shouldDisplayNewIndicator = this.props.report.newMarkerSequenceNumber > 0
+            && item.action.sequenceNumber === this.props.report.newMarkerSequenceNumber;
```

Comment:
Making sure I understand this, it should be `>=` or `===`?

Reply:
Can you ask this question another way?

I believe the left side of the condition is checking to make sure we do not have `0` since that would show a new line indicator at the bottom of the chat and it's supposed to go above _at least one_ chat message.

Reply:
I was asking this for this part of the condition `item.action.sequenceNumber === this.props.report.newMarkerSequenceNumber`. From my understanding, I thought for a current item to be considered new its sequence number must be greater than the `newMarkerSequenceNumber` or it should be equal?

Reply:
Hmm sorry, I still don't understand what it is you're asking. It sounds like you concluded that the current item's sequence number would need to be greater than the "new marker sequence number"? I'm not sure how you got that though. Let me know if you have a specific question and try to add as much context as possible if you can.